plutus-scb/src/Plutus/SCB/App.hs

```diff
@@ -113,23 +114,25 @@ type AppBackend m =
 runAppBackend ::
     forall m a.
     ( MonadIO m
-    , MonadLogger m
     , MonadUnliftIO m
+    , MonadLogger m
+    , MonadCatch m
     )
-    => Env
-    -> Eff (AppBackend m) a
+    => CM.Configuration -- ^ Logging / monitoring configuration
+    -> Trace m SCBLogMsg -- ^ Top-level tracer
+    -> Env -- ^ Client config
+    -> Eff (AppBackend m) a -- ^ Action
     -> m (Either SCBError a)
-runAppBackend e@Env{dbConnection, nodeClientEnv, walletClientEnv, signingProcessEnv, chainIndexEnv} =
+runAppBackend config trace e@Env{dbConnection, nodeClientEnv, walletClientEnv, signingProcessEnv, chainIndexEnv} =
     runM
     . runReader e
     . runReader dbConnection
-    . runStderrLog
-    . handleObserveLog
-    . renderLogMessages
-    . renderLogMessages
-    . renderLogMessages
-    . renderLogMessages
-    . renderLogMessages
+    . handleObserveTrace config trace
+    . handleLogMsgTraceMap SCoreMsg trace
```

Comment:
I won't argue that this should take prisms ;)

Reply:
Then I won't respond by pointing out the fact that functions are more general than prisms and I am merely applying the principle of "be liberal in what you accept and strict in what you produce" ;)